Fabric.Identity.API/scripts/Install-Identity-Utilities.psm1

```diff
@@ -0,0 +1,621 @@
+    }else{
+        Write-DosMessage -Level "Error" -Message "Could not find file or directory $zipPackage, please verify that the zipPackage configuration setting in install.config is the path to a valid zip file that exists."
+        throw
+    }
+}
+
+function Install-DotNetCoreIfNeeded([string] $version, [string] $downloadUrl){
+    if(!(Test-PrerequisiteExact "*.NET Core*Windows Server Hosting*" $version))
+    {    
+        try{
+            Write-DosMessage -Level "Information" -Message "Windows Server Hosting Bundle version $version not installed...installing version $version"        
+            Invoke-WebRequest -Uri $downloadUrl -OutFile $env:Temp\bundle.exe
+            Start-Process $env:Temp\bundle.exe -Wait -ArgumentList '/quiet /install'
+            Restart-W3SVC
+        }catch{
+            Write-DosMessage -Level "Error" -Message "Could not install .NET Windows Server Hosting bundle. Please install the hosting bundle before proceeding. $downloadUrl"
+            throw
+        }
+        try{
+            Remove-Item $env:Temp\bundle.exe
+        }catch{
+            $e = $_.Exception
+            Write-DosMessage -Level "Warning" -Message "Unable to remove temporary dowload file for server hosting bundle exe" 
+            Write-DosMessage -Level "Warning" -Message  $e.Message
+        }
+
+    }else{
+        Write-DosMessage -Level "Information" -Message  ".NET Core Windows Server Hosting Bundle (v$version) installed and meets expectations."
+    }
+}
+
+function Get-IISWebSiteForInstall([string] $selectedSiteName, [bool] $quiet){
+    try{
+        $sites = Get-ChildItem IIS:\Sites
+        if($quiet -eq $true){
+            $selectedSite = $sites | Where-Object { $_.Name -eq $selectedSiteName }
+        }else{
+            if($sites -is [array]){
+                $sites |
+                    ForEach-Object {New-Object PSCustomObject -Property @{
+                        'Id'=$_.id;
+                        'Name'=$_.name;
+                        'Physical Path'=[System.Environment]::ExpandEnvironmentVariables($_.physicalPath);
+                        'Bindings'=$_.bindings;
+                    };} |
+                    Format-Table Id,Name,'Physical Path',Bindings -AutoSize | Out-Host
+
+                $selectedSiteId = Read-Host "Select a web site by Id"
+                $selectedSite = $sites[$selectedSiteId - 1]
+            }else{
+                $selectedSite = $sites
+            }
+        }
+        if($null -eq $selectedSite){
+            throw "Could not find selected site."
+        }
+        if($selectedSite.Name){ Add-InstallationSetting "identity" "siteName" $selectedSite.Name | Out-Null }
+
+        return $selectedSite
+
+    }catch{
+        Write-DosMessage -Level "Error" -Message "Could not select a website."
+        throw
+    }
+}
+
+function Get-Certificates([string] $primarySigningCertificateThumbprint, [string] $encryptionCertificateThumbprint, [bool] $quiet){
```

Comment:
should we filter this to only SSL certs?

Reply:
Yes, we can look at ways to filter this list down.